plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/delete/DeletionVectors.java

```diff
@@ -0,0 +1,270 @@
+    {
+        if (storageType.equals(UUID_MARKER)) {
+            TrinoInputFile inputFile = fileSystem.newInputFile(location.appendPath(toFileName(pathOrInlineDv)));
+            ByteBuffer buffer = readDeletionVector(inputFile, offset.orElseThrow(), sizeInBytes);
+            RoaringBitmap[] bitmaps = deserializeDeletionVectors(buffer);
+            if (bitmaps.length != cardinality) {
+                // Don't throw an exception because Databricks may report the wrong cardinality when there are many deleted rows
+                log.debug("The number of deleted rows expects %s but got %s", cardinality, bitmaps.length);
+            }
+            return bitmaps;
+        }
+        else if (storageType.equals(INLINE_MARKER) || storageType.equals(PATH_MARKER)) {
+            throw new TrinoException(NOT_SUPPORTED, "Unsupported storage type for deletion vector: " + storageType);
+        }
+        throw new IllegalArgumentException("Unexpected storage type: " + storageType);
+    }
+
+    public static String toFileName(String pathOrInlineDv)
+    {
+        int randomPrefixLength = pathOrInlineDv.length() - Base85Codec.ENCODED_UUID_LENGTH;
+        String randomPrefix = pathOrInlineDv.substring(0, randomPrefixLength);
+        checkArgument(ALPHANUMERIC.matchesAllOf(randomPrefix), "Random prefix must be alphanumeric: %s", randomPrefix);
+        String prefix = randomPrefix.isEmpty() ? "" : randomPrefix + "/";
+        String encodedUuid = pathOrInlineDv.substring(randomPrefixLength);
+        UUID uuid = Base85Codec.decodeUuid(encodedUuid);
+        return "%sdeletion_vector_%s.bin".formatted(prefix, uuid);
+    }
+
+    public static ByteBuffer readDeletionVector(TrinoInputFile inputFile, int offset, int expectedSize)
+            throws IOException
+    {
+        byte[] bytes = new byte[expectedSize];
+        TrinoDataInputStream inputStream = new TrinoDataInputStream(inputFile.newStream());
+        inputStream.seek(offset);
+        int actualSize = inputStream.readInt();
+        if (actualSize != expectedSize) {
+            // TODO: Investigate why these size differ
+            log.warn("The size of deletion vector %s expects %s but got %s", inputFile.location(), expectedSize, actualSize);
+        }
+        inputStream.readFully(bytes);
+        return ByteBuffer.wrap(bytes).order(LITTLE_ENDIAN);
+    }
+
+    public static boolean contains(RoaringBitmap[] bitmaps, long value)
+    {
+        int high = highBytes(value);
+        if (high >= bitmaps.length) {
+            return false;
+        }
+        RoaringBitmap highBitmap = bitmaps[high];
+        int low = lowBytes(value);
+        return highBitmap.contains(low);
+    }
+
+    private static int highBytes(long value)
+    {
+        return toIntExact(value >> 32);
+    }
+
+    private static int lowBytes(long value)
+    {
+        return toIntExact(value);
+    }
+
+    public static RoaringBitmap[] deserializeDeletionVectors(ByteBuffer buffer)
```

Comment:
add unit tests.
(can be own class, prep commit)

Reply:
This method is tested via `readDeletionVectors` method. I will change to private.